server/internal/server_test.go

```diff
@@ -33,6 +35,16 @@ func (a *mockAdapter[Config]) GetPage(ctx context.Context, request *framework.Re
 }
 
 func TestServer_GetPage(t *testing.T) {
+	path := "./TOKENS"
+	if err := os.Setenv("AUTH_TOKENS_PATH", path); err != nil {
```

Comment:
How are you envisioning we test `GetPage` without relying on this env being set since it's required for auth?

Are you thinking `validateAuthenticationToken` should be restructured to load the list of valid auth tokens on init and use that?

One concern with making that change is that we no longer have test coverage on loading the secrets from the file, unless of course we add another test that relies on an env.

Reply:
> Are you thinking validateAuthenticationToken should be restructured to load the list of valid auth tokens on init and use that?

No, because the tokens can be modified at runtime.

Instead, I was thinking of reading the tokens at startup, and then set up a file watcher to reload them _only_ when the file is modified.

Reply:
Removed the env dependency from the `TestServer_GetPage` test. 

Since this is a framework and does not have a clear entry point, I added the logic to read the tokens and startup the file watcher to the function that initializes a new AdapterServer as this seems like the clear entrypoint to me (this will always be called first and is required to use the framework). I've also added a field to store this tokens to the `internal.Server` type. 

A few important notes:
- With the current implementation I'm going with the assumption that the service should panic if the file watcher  is unable to function properly to prevent a situation where we're unable to retrieve an updated list of tokens (and therefore potentially provide unintended access)
- I'm still unable to think of a way to test the logic where we're loading the tokens without relying on an environment variable, since this logic relies on the `AUTH_TOKENS_PATH` env to be set. IMO this logic needs to have test coverage - so are you fine with me adding a serial test for `New` (api_adapter_v1.AdapterServer) using the built in https://pkg.go.dev/testing#B.Setenv function?

Reply:
> With the current implementation I'm going with the assumption that the service should panic if the file watcher is unable to function properly

+1

Reply:
The env variable (the path to the tokens file) cannot change.
So I would just get the filename from the env variable in `main` and just pass the filename around as a parameter.

Reply:
When you say `main` are you referring to the `main` function in `adapter-sgnl` (e.g. require the adapter implementation using the framework to provide this)? There's no `main` file / function in `adapter-framework` I'm seeing since this is just being used as a library. We could use `init()`, but it seems to me like there are some problems with this approach.